example/src/release_tests/touchables/index.tsx

```diff
@@ -33,7 +33,9 @@ const renderSampleBox = (color?: string) => (
   />
 );
 
-const toReactNativeTouchable = (touchable: React.ComponentType<unknown>) => {
+const toReactNativeTouchable = (
+  touchable: React.ComponentType<unknown>
+): React.ComponentType<any> => {
```

Comment:
Is there a way to avoid `any`?

Reply:
Also, why in some functions we remove return types, while in other we add them?

Reply:
>why in some functions we remove return types

I removed all instances of `ViewStyle`, as it's incompatible between `0.75` and `0.74`, in all instances where it's been removed, the type has been automatically inferred.

>while in other we add them?

Which ones are you referring to?
I can only see types being either removed, or replaced by generic ones where they couldn't've been inferred.

>Is there a way to avoid any?

Unfortunately I couldn't find a way to.
`unknown` and `never` won't work, and i couldn't find another type which would prevent the errors being thrown. The type also cannot be automatically inferred in this instance.
This issue is a result of `RN`'s `TouchableHighlight` and `TouchableOpacity` being rewritten to function-style components in `0.75` while `TouchableNativeFeedback` and `TouchableWithoutFeedback` remain class components.

Reply:
>To this one 😉 Previously there was no type annotation for return value.

Right, sorry, `React.ComponentType<any>` in this case means that a react component with `any` props will be returned. I'll have to give it a second thought. I'm not sure if the new function-style components it is now returning can be used in the same way the old class components could've been used.

Reply:
fixed in [273661f](https://github.com/software-mansion/react-native-gesture-handler/pull/3204/commits/273661f97aae6b0e53fb8d700767e79ccc9a683e)

Reply:
What I meant was to ask if TypeScript has a problem with type inference in that case?

Reply:
>What I meant was to ask if TypeScript has a problem with type inference in that case?

It does. Here's the error that's thrown when using the element returned by this function:
![image](https://github.com/user-attachments/assets/5a3fa1db-afed-4269-9ee8-e404e825137d)

Also, I tested all the problematic components - `RNTouchableOpacity` and `RNTouchableHighlight` - which are the reason for those type errors, and they function as intended. All props are properly applied, the components are rendered.